src/bst.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-o
+from collections import deque
+
+class Node(object):
+    """Building our Node class."""
+
+    def __init__(self, val, left=None, right=None):
+        """Class implements a node."""
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+class BST(object):
+    """Class implements a binary search tree data structure in Python."""
+
+    def __init__(self, iterable=None):
+        """Init method of bst class."""
+        self.root = None
+        self.length = 0
+        self.depth_left = 0
+        self.depth_right = 0
+        if iterable is None:
+            self.root = None
+            return
+        for value in iterable:
+            self.insert(value)
+
+    def insert(self, val, node=None):
+        """Insert new node into bst."""
+        new_node = Node(val)
+        if not self.root:
+            self.root = new_node
+            self.length = 1
+        if node is None:
+            node = self.root
+        if val > node.val:
+            if not node.right:
+                node.right = new_node
+                self.length += 1
+            else:
+                self.insert(val, node.right)
+        elif val < node.val:
+            if not node.left:
+                node.left = new_node
+                self.length += 1
+            else:
+                self.insert(val, node.left)
+
+    def contains(self, val):
+        """Check if value exists in bst."""
+        current_node = self.root
+        if current_node is None:
+            return False
+        if val == current_node.val:
+            return True
+        while True:
+            if val > current_node.val and current_node.right is not None:
```

Comment:
Good job using the power of your BST to keep your `contains` method at O(logn) time.